core/src/main/java/org/bouncycastle/asn1/DateUtil.java

```diff
@@ -64,7 +64,7 @@ static Date epochAdjust(Date date)
                 localeCache.put(locale, adj);
             }
 
-            if (adj != ZERO)
+            if (!adj.equals(ZERO))
```

Comment:
This is a false positive. Better remove this change and let us review the file separately.